pdns/syncres.hh

```diff
@@ -639,6 +639,12 @@ public:
     d_skipCNAMECheck = skip;
   }
 
+  void setQuerySource(const ComboAddress& requestor, boost::optional<const EDNSSubnetOpts&> incomingECS) {
```

Comment:
I'm not sure this is a good idea to create a function in 4.1.x that does something quite different from the one in master. It's clearly very useful when backporting the `SyncRes` unit tests, but I'm worried it might hide the fact that the handling of incoming ECS values is very different in 4.1.x during a backport. I would not worry so much about that normally, but we have been backporting a lot of things to 4.1.x recently..
Perhaps we could have a wrapper in the `SyncRes` unit tests but not make it part of the `SyncRes` class itself to prevent any confusion?

Reply:
Yes, better indeed. Will do.